File: data/compilers_comparison/src/examples/second_example/src/func18.c
#include "second_example.h" 
array_t* func18(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array189 = func19(&params0, rng(), loopsFactor);
      DEBUG_RETURN(array189->id);
      free(params0.data);
      array189->refC--;
      if(array189->refC == 0) {
         free(array189->data);
         free(array189);
         DEBUG_FREE(array189->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array199 = func20(&params0, rng(), loopsFactor);
      DEBUG_RETURN(array199->id);
      free(params0.data);
      array199->refC--;
      if(array199->refC == 0) {
         free(array199->data);
         free(array199);
         DEBUG_FREE(array199->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array204 = func21(&params0, rng(), loopsFactor);
   DEBUG_RETURN(array204->id);
   free(params0.data);
   return array204;
}

File: data/multilang_comparison/program_nim/src/func18.nim
proc func18*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array221: ptr ArrayT = func19(addr params0, getPath(), loopsFactor)
      debugReturn(array221.id)
      dealloc(params0.data)
      dec array221.refC
      if array221.refC == 0:
         dealloc(array221.data)
         dealloc(array221)
         debugFree(array221.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array231: ptr ArrayT = func20(addr params0, getPath(), loopsFactor)
      debugReturn(array231.id)
      dealloc(params0.data)
      dec array231.refC
      if array231.refC == 0:
         dealloc(array231.data)
         dealloc(array231)
         debugFree(array231.id)
   var params0: ArrayTParam
   params0.size = 0
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   var array236: ptr ArrayT = func21(addr params0, getPath(), loopsFactor)
   debugReturn(array236.id)
   dealloc(params0.data)
   return array236

